resources/session03/wsgi/calculator.py

```diff
@@ -0,0 +1,79 @@
+
+
+def num(s):
+    # return some sort of number from a string
+    try:
+        return int(s)
+    except ValueError:
+        return float(s)
+
+def multiply(arg1, arg2):
+    return("{} * {} = {}".format(arg1, arg2, arg1*arg2))
+
+
+def add(arg1, arg2):
+    return("{} + {} = {}".format(arg1, arg2, arg1+arg2))
+
+
+def subtract(arg1, arg2):
+    return("{} - {} = {}".format(arg1, arg2, arg1-arg2))
+
+
+def divide(arg1, arg2):
+    return("{} / {} = {}".format(arg1, arg2, arg1/arg2))
+
+
+
+operator = {"add": add, "subtract": subtract, "multiply": multiply,
+        "divide": divide}
+
+
+def resolve_path(path):
+    '''
+    Try to find one of the operators in the path, then extract the
+    arguments from the path
+
+    Check for valid operators and operands
+
+    Return function based on the operator and return operands in a list
+    '''
+    matchpath = path.lstrip('/')
+    parts = matchpath.split('/')
+    try:
+        func = operator[parts[0]]
+    except KeyError:
+        raise NameError
+    arg1 = num(parts[1])
+    arg2 = num(parts[2])
```

Comment:
Your validation function above is still a bit fragile, think about what that means for this code here.